src/ArtisanApartmentCommands.php

```diff
@@ -0,0 +1,176 @@
+use BuildEmpire\Apartment\Schema;
+use Illuminate\Support\Facades\Schema as LumenSchema;
+use Carbon\Carbon;
+use BuildEmpire\Apartment\Helpers\ApartmentHelpers;
+
+class ArtisanApartmentCommands
+{
+    const ARTISAN = 1;
+    const WEB = 2;
+    const PUBLIC_SCHEMA = 'public';
+
+    protected $schema = false;
+
+    /**
+     * ArtisanApartmentCommands constructor.
+     * @param \BuildEmpire\Apartment\Schema $schema
+     */
+    public function __construct(Schema $schema)
+    {
+        $this->schema = $schema;
+    }
+
+    /**
+     * Make an apartment's schema and run any existing apartment migrations.
+     *
+     * @param $schemaName
+     * @throws SchemaAlreadyExistsException
+     * @throws SchemaCannotBePublicException
+     * @throws SchemaNameNotValidException
+     */
+    public function makeSchema($schemaName)
+    {
+
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+
+        if ($this->schema->doesSchemaExist($schemaName)) {
+            throw new SchemaAlreadyExistsException('The apartment ' . $schemaName . ' already exists.');
+        }
+
+        if ($this->isSchemaPublic($schemaName)) {
+            throw new SchemaCannotBePublicException('The apartment name cannot be ' . self::PUBLIC_SCHEMA);
+        }
+
+        app('db')->transaction(function () use ($schemaName) {
+
+            $this->createSchema($schemaName);
+
+            $this->createSchemaMigrationTable($schemaName);
+
+            $this->createSchemaMetadata($schemaName);
+
+            $this->updateSchemaMetadata($schemaName, self::ARTISAN, Carbon::now()->getTimestamp());
+
+            Migrator::runMigrations();
+        });
+    }
+
+    /**
+     * Drop an existing schema and all it's data with cascade enabled.
+     *
+     * @param $schemaName
+     * @throws SchemaDoesntExistException
+     * @throws SchemaNameNotValidException
+     */
+    public function dropSchema($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+
+        if (!$this->schema->doesSchemaExist($schemaName)) {
+            throw new SchemaDoesntExistException('The apartment ' . $schemaName . ' does not exist.');
+        }
+
+        app('db')->transaction(function () use ($schemaName) {
+            $this->deleteSchema($schemaName);
+        });
+    }
+
+    /**
+     * Create the schema.
+     *
+     * @param $schemaName
+     * @throws SchemaNameNotValidException
+     */
+    protected function createSchema($schemaName)
+    {
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
```

Comment:
It's slightly inconsistent that some of the protected methods have these checks and some don't - given this class isn't supposed to be extended (?) should the checks actually only exist in the public methods (somewhat contrary to my comment above which suggested "lowest" point, though the lowest point is with respect to intended consumers)?